00_Base/src/services/credentials.service.ts

```diff
@@ -474,6 +477,119 @@ export class CredentialsService {
     return updatedClientInformation;
   }
 
+  async generateCredentialsTokenA(
```

Comment:
I might be missing something but shouldn't the logic for `generateCredentialsTokenA` and `registerCredentialsTokenA` be very similar?

The only difference should be that `generateCredentialsTokenA` generates a new token while `registerCredentialsTokenA` uses the token provided in the request.

If so, then I would expect the logic in `generateCredentialsTokenA` to be as simple as:
1. Generate token
2. Make call to `registerCredentialsTokenA`

Reply:
The registration process is symmetric (refer OCPI 7.1.1. Registration). OCPI can be the sender or the receiver.

When OCPI is the sender and MSP is the receiver, we call `registerCredentialsTokenA`. The token A is from MSP and is a temp token which is only valid during the registration process(MSP sends us token A by email or any other safe way). In the `registerCredentialsTokenA`, we use this token A to get MSP side version, version details and send them our token B. In the response, MSP send us the client roles as well as formal token C which replaces the temp token A. The registration process finishes.

When OCPI is the receiver and MSP is the sender,  we call `generateCredentialsTokenA` to generate our side temp token A and prepare server side info, e.g., server roles, versions and details. Then we send this token A to MSP also by emails or other ways. This time MSP execute the registration like we do in the `registerCredentialsTokenA`. 

So we can not simply do generate token and call `registerCredentialsTokenA` together. They serves for different cases as sender and receiver.